packages/dumb/src/provider/trends.provider.hook.tsx

```diff
@@ -25,21 +25,34 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-import { useState } from 'react'
+import { useRef, useState } from 'react'
 import { type TrendsDisplayFlags } from '../models/trends-display-flags.model'
 import { DisplayFlag } from '../models/enums/display-flag.enum'
 import { type CbgDateTrace } from '../models/cbg-date-trace.model'
-import { type CbgPositionData } from '../models/cbg-position-data.model'
+import {
+  type CbgDateTraceYPositions,
+  type CbgPositionData,
+  type CbgSliceYPositions
+} from '../models/cbg-position-data.model'
 import { type FocusedCbgDateTrace } from '../models/focused-cbg-date-trace.model'
+import { type RangeSegmentSlice } from '../models/enums/range-segment.enum'
+import { type CbgSlice } from '../models/cbg-slice.model'
+import { type FocusedCbgSlice } from '../models/focused-cbg-slice.model'
 
 export interface TrendsContextResult {
   displayFlags: TrendsDisplayFlags
-  focusCbgDateTrace: (cbgDateTrace: CbgDateTrace, position: CbgPositionData) => void
+  focusCbgDateTrace: (cbgDateTrace: CbgDateTrace, position: CbgPositionData<CbgDateTraceYPositions>) => void
   focusedCbgDateTrace: FocusedCbgDateTrace | undefined
+  focusCbgSlice: (cbgSlice: CbgSlice, position: CbgPositionData<CbgSliceYPositions>, keys: RangeSegmentSlice[]) => void
+  focusedCbgSlice: FocusedCbgSlice | undefined
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches to further communicate your intent.  -->
**🎯 suggestion (non-blocking):** ‏
```suggestion
  focusedCbgDateTrace?: FocusedCbgDateTrace
```

Reply:
Smart! 😎